src/main/java/Domain/ShoppingCart.java

```diff
@@ -1,5 +1,54 @@
 package Domain;
 
+import java.util.List;
+import java.util.ArrayList;
+
+import Domain.Exceptions.ItemOutOfStockExepction;
+
 public class ShoppingCart {
+    private List<ShoppingBasket> _shoppingBaskets;
+
+    public ShoppingCart() {
+        _shoppingBaskets = new ArrayList<>();
+    }
+
+    /*
+     * Go thorugh the list of baskets to buy and purchase them.
+     * If an exception is thrown, cancel the purchase of all the baskets that were bought.
+     * This function only updates the item's stock. 
+     */
+    public void purchaseCart(User buyer, List<Integer> busketsToBuy) {
```

Comment:
where is the connection to payment and shipping? if not implemented yet pls add //TODO: .... for later to do not forget